### PR TITLE
Do not update running list when hidden

### PR DESCRIPTION
**Pull request by krassowski**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/13676

## Code changes

Filter our signal triggering updates if the section widget (or more accurately any of its parents) is hidden.

## User-facing changes

Slightly better performance on consecutive tab switch when running section is hidden.

Comparison using a workspace with 120 tabs open (exaggerating a bit for ease of visualisation - most workspaces are smaller).

### Before/after when running section active

Screenshot from 2023-03-12 13-34-25

### After when running section is hidden

Screenshot from 2023-03-12 13-35-06

While the performance benefit in this PR is modest, having it in will also help future profiling effort as the cleaner the timeline the easier it is to notice the things we can optimise.

## Backwards-incompatible changes

None

### CODE REVIEW

File: packages/running/src/index.tsx [hunk 1]
```diff
@@ -250,6 +251,90 @@ function List(props: {
...
+      translator?: ITranslator;
+    }
+  ) {
+    super();
+    _options.manager.runningChanged.connect(this._emitUpdate, this);
+  }
+
+  dispose() {
+    super.dispose();
+    this._options.manager.runningChanged.disconnect(this._emitUpdate, this);
```
Comment (fcollonval) [inline packages/running/src/index.tsx:269]: The undo action should mirror the action order.

```suggestion
    this._options.manager.runningChanged.disconnect(this._emitUpdate, this);
    super.dispose();
```

File: packages/running/src/index.tsx [hunk 2]
```diff
@@ -250,6 +251,90 @@ function List(props: {
...
+  }
+
+  onBeforeShow() {
```
Comment (fcollonval) [inline packages/running/src/index.tsx:272]: Let's keep the default visibility

```suggestion
  protected onBeforeShow(msg: Message): void {
    super.onBeforeShow(msg);
```

File: packages/running/src/index.tsx [hunk 3]
```diff
@@ -250,6 +251,90 @@ function List(props: {
...
+    this._update.emit();
+  }
+
+  render() {
```
Comment (fcollonval) [inline packages/running/src/index.tsx:276]:
```suggestion
  render(): JSX.Element {
```

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @krassowski

I have small comments and a general design question:

Should we make this (aka no update if hidden) part of `UseSignal`?

> I have no strong feelings moving forward with the code as is.

**Comment by krassowski** [comment]
> Should we make this (aka no update if hidden) part of `UseSignal`?

That would be great but how could we do that? It seems to me that `UseSignal` does not know about the parent widget so we would need to add an argument passing `this` with reference to lumino widget which seems a bit problematic? We can discuss in an separate issue especially if this requires deeper changes.

**Review by fcollonval — APPROVED** [review]
Thanks @krassowski

**Pull request merged**